modules/host/host.go

```diff
@@ -105,8 +106,19 @@ func New(cs *consensus.ConsensusSet, hdb modules.HostDB, tpool modules.Transacti
 	if err != nil {
 		return nil, err
 	}
+	host, err := modules.IGD.ExternalIP()
+	if err != nil {
+		// log error? should be fatal
+		host = "::1"
+	}
 	_, port, _ := net.SplitHostPort(h.listener.Addr().String())
-	h.myAddr = modules.NetAddress(net.JoinHostPort(modules.ExternalIP, port))
+	h.myAddr = modules.NetAddress(net.JoinHostPort(host, port))
+
+	// Forward port, if possible
+	portInt, _ := strconv.Atoi(port)
+	if portErr := modules.IGD.Forward(uint16(portInt), "Sia Host"); portErr != nil {
+		// log error? should be fatal if err != modules.ErrNoUPnP
```

Comment:
let upnp fail silently (with a logging error), and add a logger to the host to record the error.

any function that requires a forwarded port (such as announcing, or even setting the host to accept contracts) should return an error if the 'have I forwarded my ports' test hasn't been passed.